File: app/src/main/java/com/luizroseiro/testeandroidv2/views/activities/MainActivity.java
package com.luizroseiro.testeandroidv2.views.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.luizroseiro.testeandroidv2.R;
import com.luizroseiro.testeandroidv2.services.DataService;
import com.luizroseiro.testeandroidv2.utils.Utils;
import com.luizroseiro.testeandroidv2.views.fragments.LoginFragment;
import com.luizroseiro.testeandroidv2.views.fragments.MainFragment;

public class MainActivity extends AppCompatActivity {

    private static MainActivity mainActivity;
    public static MainActivity getMainActivity() {
        return mainActivity;
    }
    public static void setMainActivity(MainActivity mainActivity) {
        MainActivity.mainActivity = mainActivity;
    }

    public static DataService dataService;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setMainActivity(this);
        dataService = new DataService(this);

        if (dataService.isLoggedIn())
            Utils.replaceFragment(R.id.container_main, new MainFragment());
        else
            Utils.replaceFragment(R.id.container_main, new LoginFragment());

    }

}

File: BankApp/app/src/main/java/com/accenture/bankapp/MainActivity.kt
package com.accenture.bankapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.accenture.bankapp.screens.dashboard.DashboardFragment
import com.accenture.bankapp.screens.dashboard.DashboardFragmentListener
import com.accenture.bankapp.screens.login.LoginFragment
import com.accenture.bankapp.screens.login.LoginFragmentListener
import com.accenture.bankapp.utils.transact
import timber.log.Timber
import timber.log.Timber.DebugTree


class MainActivity : AppCompatActivity(), LoginFragmentListener, DashboardFragmentListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        if (BuildConfig.DEBUG) {
            Timber.plant(DebugTree())
        }

        Timber.i("onCreate: Creating the Main Activity")

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        showFragment(LoginFragment())
    }

    override fun startDashboardFragment(dashboardFragment: DashboardFragment) {
        Timber.i("startDashboardFragment: Starting Dashboard Fragment")

        showFragment(dashboardFragment)
    }

    override fun startLoginFragment(loginFragment: LoginFragment) {
        Timber.i("startLoginFragment: Starting Login Fragment")

        showFragment(loginFragment)
    }

    private fun showFragment(fragment: Fragment) {
        Timber.i("showFragment: Showing the ${fragment::class.simpleName}")

        transact {
            replace(R.id.layoutContainer, fragment)
            setCustomAnimations(R.anim.abc_fade_in, R.anim.abc_fade_out)
        }
    }
}